nilearn/regions/hierarchical_kmeans_clustering.py

```diff
@@ -0,0 +1,277 @@
+import numpy as np
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.base import TransformerMixin, ClusterMixin
+from sklearn.base import BaseEstimator
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+import warnings
+
+
+def _remove_empty_labels(labels):
+    '''Remove empty values label values from labels list'''
+    vals = np.unique(labels)
+    inverse_vals = - np.ones(labels.max() + 1).astype(np.int)
+    inverse_vals[vals] = np.arange(len(vals))
+    return inverse_vals[labels]
+
+
+def _adjust_small_clusters(array, n_clusters):
+    '''Takes a ndarray of floats summing to n_clusters and try to round it while
+    enforcing rounded array still sum to n_clusters and every element is at
+    least 1.
+    '''
+
+    array_round = np.round(array).astype(int)
+    array_round = np.maximum(array_round, np.ones(array.size))
+    if np.sum(array_round) == n_clusters:
```

Comment:
I'd rather not for code readability. I reordered the cases to make the most frequent come first but if it had to be done 3 times, it would still only be summing a list of length << 100